src/App.tsx

```diff
@@ -8,37 +8,89 @@ import { PostsList } from './components/PostsList';
+import { getUsers, getPosts } from './api';
+import { Post } from './types/Post';
+import { ErrorTypes } from './types/ErrorAction';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorType, setErrorType] = useState<ErrorTypes | null>(null);
+  const [isSelectedUser, setSelectedUser] = useState(false);
+  const [selectPost, setSelectPost] = useState<Post | null>(null);
+
+  useEffect(() => {
+    getUsers()
+      .then(result => {
+        setUsers(result);
+      })
+      .catch(() => {
+        setErrorType(ErrorTypes.USERS);
+      });
+  }, []);
+
+  const getPostsById = (id:number) => {
+    setIsLoading(true);
+    getPosts(id)
+      .then(result => {
+        setPosts(result);
+        setSelectedUser(true);
+      })
+      .catch(() => {
+        setErrorType(ErrorTypes.POSTS);
+      })
+      .finally(() => setIsLoading(false));
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  getPostsById={getPostsById}
+                  setSelectPost={setSelectPost}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!isSelectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
+                {errorType === ErrorTypes.USERS && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {ErrorTypes.USERS.toString()}
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {errorType === ErrorTypes.POSTS && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    {ErrorTypes.POSTS.toString()}
+                  </div>
+                )}
 
-                <PostsList />
+                {isLoading
+                  ? <Loader />
+                  : isSelectedUser && (
```

Comment:
better to use `Loader` separately and only check `isLoading`

and here you can check `isSelectedUser` and just show `PostList` or `No user selected`